modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/Diagram.js

```diff
@@ -64,48 +70,120 @@ const Diagram = ({
 		radius: newPinSettings.defaultRadius,
 	});
 
-	useEffect(() => {
-		fetch(`${pinsEndpoint}${productId}/pins`)
-			.then((response) => response.json())
-			.then((jsonResponse) => {
-				const loadedPins = jsonResponse.items.map((item) => ({
-					cx: item.positionX,
-					cy: item.positionY,
-					id: item.id,
-					label: item.number,
-				}));
+	const loadPins = useCallback(
+		() =>
+			fetch(`${pinsEndpoint}${PRODUCTS}/${productId}/${PINS}`, {
+				headers: new Headers({
+					Accept: 'application/json',
+					'Content-Type': 'application/json',
+				}),
+			})
+				.then((response) => response.json())
+				.then((jsonResponse) => {
+					const loadedPins = jsonResponse.items.map((item) => ({
+						cx: item.positionX,
+						cy: item.positionY,
+						id: item.id,
+						label: item.number,
+					}));
 
-				setCpins(loadedPins);
-			});
-	}, [pinsEndpoint, productId]);
+					setCpins(loadedPins);
+				}),
+		[pinsEndpoint, productId]
+	);
 
-	useEffect(() => {
-		if (!showTooltip.tooltip) {
-			const newCPinState = cPins?.map((element) => {
-				if (element.id === showTooltip.details.id) {
-					return {
-						cx: cPins[element.id].cx,
-						cy: cPins[element.id].cy,
-						draggable: cPins[element.id].draggable,
-						fill: cPins[element.id].fill,
-						id: showTooltip.details.id,
-						label: showTooltip.details.label,
-						linked_to_sku: showTooltip.details.linked_to_sku,
-						quantity: showTooltip.details.quantity,
-						r: cPins[element.id].r,
-						sku: showTooltip.details.sku,
-					};
-				}
-				else {
-					return element;
-				}
+	const deletePin = (node) => {
+		fetch(`${pinsEndpoint}${PINS}/${node.id}`, {
+			headers: new Headers({
+				Accept: 'application/json',
+				'Content-Type': 'application/json',
+			}),
```

Comment:
Rather than re-declaring these over and over, can you just put this as a constant at the top of the file?

```js
const HEADERS = new Headers({
	Accept: 'application/json',
	'Content-Type': 'application/json',
})
```